vcap_utils/vcap_utils/backends/base_openvino.py

```diff
@@ -35,34 +35,128 @@ def __init__(self, model_xml: bytes,
+            # but load onto 'CPU', which is a safe choice.
+            msg = f"Failed to load {self.__class__} onto device " \
+                  f"{device_name}. Error: '{e}'. " \
+                  f"Trying again with device_name='CPU'"
+            logging.error(msg)
+            self.__init__(
+                model_xml=model_xml,
+                weights_bin=weights_bin,
+                device_name="CPU",
+                ie_core=ie_core)
+            return
 
         # Pull out a couple useful constants
-        self.InferRequestStatusCode = StatusCode
         self.input_blob_names: List[str] = list(self.net.inputs.keys())
         self.output_blob_names: List[str] = list(self.net.outputs.keys())
 
+        # For running threaded requests to the network
+        self._StatusCode = StatusCode
+        self._get_free_request_lock = RLock()
+        self._request_free_events: List[Event] = [
+            Event() for _ in self.exec_net.requests]
+        """This list corresponds 1:1 with request ID's from OpenVINO. 
+        The Events are used to make sure that the request isn't in the middle
+        of running a callback while another thread attempts to start another
+        async request. 
+        
+        The events start as 'set' to indicate they are free to run
+        """
+        for request_free in self._request_free_events:
+            request_free.set()
+
+        # For keeping track of the workload for this backend
+        self._total_requests: int = len(self.exec_net.requests)
+        self._num_ongoing_requests: int = 0
+        self._num_ongoing_requests_lock: RLock = RLock()
+
+    @property
+    def workload(self) -> float:
+        """Returns the percent saturation of this backend. The backend is
+        at max 'efficiency' once the number of ongoing requests is equal to
+        or over number of exec_net.requests
+
+        This won't affect much unless a custom DeviceMapper filter is used that
+        allows multiple Backends to be loaded, eg, a backend for CPU and a
+        backend for HDDL. In those cases, this workload measurement will be used
+        heavily to decide which backend is busier.
+        """
+        return self._num_ongoing_requests / self._total_requests
+
+    def send_to_batch(self, input_data: OV_INPUT_TYPE) -> Queue:
+        """Efficiently send the input to be inferenced by the network
+        :param input_data: Input to the network
+        :returns: A queue that will yield 1 result, the output from the network
+        """
+        out_queue = Queue(maxsize=1)
+
+        with self._get_free_request_lock:
+            # Try to get at least one request
+            request_id = self.exec_net.get_idle_request_id()
+            if request_id < 0:
+                # Since there was no free request, wait for one
+                status = self.exec_net.wait(num_requests=1)
+                if status != self._StatusCode.OK:
```

Comment:
`is not` instead of `!=`

Reply:
Actually, we had this same discussion last pull request. OpenVINO doesn't return an enum, it returns an int. I have to do !=

Reply:
I don't remember this conversation 🤔, but that is sad